test/multimanifestmodule.test.ts

```diff
@@ -0,0 +1,156 @@
+            expect(spyStackCollector).calledOnce;
+        });
+    });
+
+    test('find_manifests_workspace should return null in callback if postStackAnalysisService fails', () => {
+        sandbox.stub(vscode.workspace, 'findFiles').resolves([{'fspath':'path/file'}]);
+        sandbox.stub(multimanifestmodule, 'form_manifests_payload').yields([{'fspath':'path/file'}]);
+        sandbox.stub(stackAnalysisServices, 'postStackAnalysisService').rejects('err');
+        multimanifestmodule.find_manifests_workspace(context, '/pom.xml', (data) => {
+            expect(data).equals(null);
+        });
+    });
+
+    test('find_manifests_workspace should return null in callback if form_manifests_payload retuns falsy', () => {
+        sandbox.stub(vscode.workspace, 'findFiles').resolves([{'fspath':'path/file'}]);
+        sandbox.stub(multimanifestmodule, 'form_manifests_payload').yields(null);
+        multimanifestmodule.find_manifests_workspace(context, '/pom.xml', (data) => {
+            expect(data).equals(null);
+        });
+    });
+
+    test('form_manifests_payload should return null in callback', () => {
+        sandbox.stub(multimanifestmodule, 'manifestFileRead').rejects('err');
+        multimanifestmodule.form_manifests_payload([{'fspath':'path/file'}], (data) => {
+            expect(data).equals(null);
+        });
+    });
+
+    test('form_manifests_payload should return form_data in callback', () => {
+        let form_data = {
+            'manifest[]': ['manifest'],
+            'filePath[]': ['path'],
+            'license[]': [{'value':'sample'}],
+            origin: 'lsp'
+        };
+        sandbox.stub(multimanifestmodule, 'manifestFileRead').resolves([{'manifest':'manifest', 'filePath': 'path', 'license': {'value': 'sample'}}]);
+        multimanifestmodule.form_manifests_payload([{'fspath':'path/file'}], (data) => {
+            expect(data).equals(form_data);
+        });
+    });
+
+    test('manifestFileRead should return error', () => {
+        sandbox.stub(fs, 'readFile').yields({'message':'err'});
+        let promiseManifestFileRead = multimanifestmodule.manifestFileRead({'fsPath': 'path'});
+        promiseManifestFileRead.catch((err)=>{
+            expect(err).equals('err');
+        });
+    });
+
+    test('manifestFileRead should return license value', () => {
+        sandbox.stub(fs, 'readFile').yields({'message':'err'}, 123);
+        let promiseManifestFileRead = multimanifestmodule.manifestFileRead({'fsPath': 'path/LICENSE'});
+        promiseManifestFileRead.then((data)=>{
+            expect(data.license.value).equals('123');
+        });
+    });
+
+    test('triggerManifestWs should return error', () => {
+        let stubAuthorize_f8_analytics =  sandbox.stub(authextension, 'authorize_f8_analytics').yields(null);
+        let promiseTriggerManifestWs = multimanifestmodule.triggerManifestWs(context, 'pom.xml', provider, previewUri);
```

Comment:
please check the expected result for this method

Reply:
The expected result is checked here

```
promiseTriggerManifestWs.catch((err)=>{
            expect(err).equals(undefined);
        });
```